initializer/test_suite_runner.go

```
@@ -100,68 +106,111 @@ func (runner TestSuiteRunner) RunTests(testNamesToRun []string) (err error) {
 	// TODO TODO TODO Support creating one network per testnet
 	_, err = dockerManager.CreateNetwork(DEFAULT_SUBNET_MASK)
 	if err != nil {
-		return stacktrace.Propagate(err, "Error in creating docker subnet for testnet.")
+		return nil, stacktrace.Propagate(err, "Error in creating docker subnet for testnet.")
 	}
 
+	// TODO break everything inside this for loop into its own function for readability
 	// TODO implement parallelism
-	// TODO implement capturing test results
+	testResults := make(map[string]TestResult)
 	for testName, test := range testsToRun {
 		logrus.Infof("Running test: %v", testName)
-		networkLoader := test.GetNetworkLoader()
-		testNetworkCfg, err := networkLoader.GetNetworkConfig()
+		networkLoader, err := test.GetNetworkLoader()
 		if err != nil {
-			return stacktrace.Propagate(err, "Unable to get network test from test provider")
+			logrus.Error("An error occurred getting the network loader:")
+			logrus.Error(err)
+			testResults[testName] = ERRORED
+			continue
 		}
 
+		builder := networks.NewServiceNetworkConfigBuilder()
+		if err := networkLoader.ConfigureNetwork(builder); err != nil {
+			logrus.Error("An error occurred configuring the test network:")
+			logrus.Error(err)
+			testResults[testName] = ERRORED
+			continue
```

WE EAT ERRORS FOR BREAKFAST

but real talk makes sense